src/swarm-engine/Swarm/Game/Step/Const.hs

```diff
@@ -1061,7 +1061,7 @@ execConst runChildProg c vs s k = do
         -- Construct the new robot and add it to the world.
         parentCtx <- use robotContext
         newRobot <-
-          zoomRobots . addTRobot (In cmd e s [FExec]) . (trobotContext .~ parentCtx) $
+          zoomRobots . fmap (robotContext .~ parentCtx) . addTRobot (In cmd e s [FExec]) $
```

Comment:
I think the reason this doesn't work is that it only modifies the `robotContext` of the `Robot` value returned by `addTRobot`, but it does not modify the original `Robot` value which `addTRobot` actually adds to the robot map (of which it then returns a copy).

Reply:
Maybe we could just have `addTRobot` take a `Maybe RobotContext` as an extra argument?  I don't know if that's actually a good idea, just random brainstorming...